lib/Doctrine/DBAL/Driver/OCI8/OCI8Connection.php

```diff
@@ -134,12 +134,16 @@ public function exec(string $statement) : int
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId(?string $name = null) : string
+    public function lastInsertId() : string
     {
-        if ($name === null) {
-            throw new OCI8Exception('The driver does not support identity columns.');
-        }
+        throw new OCI8Exception('The driver does not support identity columns.');
```

Comment:
Please see https://github.com/doctrine/dbal/pull/3759#discussion_r352711671. There should be a single `DriverException`, with factory methods for these use cases. But my previous proposal has been dismissed because @Majkl578 was already working on exceptions.